git-town fails to build on Darwin

### Steps To Reproduce
Steps to reproduce the behavior:
1. build git-town on MacOS

### Build log
```
testing: warning: no tests to run
PASS
ok      github.com/git-town/git-town/v7 0.231s [no tests to run]
# github.com/git-town/git-town/v7/src/cli
src/cli/dryrun.go:13:3: fmt.Println arg list ends with redundant newline
FAIL    github.com/git-town/git-town/v7/src/cli [build failed]
FAIL
error: builder for '/nix/store/3f6kkm3yhh3ba4a0vnanm97as6cm6265-git-town-7.7.0.drv' failed with exit code 1;
       last 10 log lines:
       > github.com/git-town/git-town/v7/test
       > Building subPackage ./test/helpers
       > running tests
       > testing: warning: no tests to run
       > PASS
       > ok    github.com/git-town/git-town/v7 0.231s [no tests to run]
       > # github.com/git-town/git-town/v7/src/cli
       > src/cli/dryrun.go:13:3: fmt.Println arg list ends with redundant newline
       > FAIL     github.com/git-town/git-town/v7/src/cli [build failed]
       > FAIL
       For full logs, run 'nix log /nix/store/3f6kkm3yhh3ba4a0vnanm97as6cm6265-git-town-7.7.0.drv'.
error: 1 dependencies of derivation '/nix/store/c1xc57p8q83b7is4qjavgbl0rqb1cg99-system-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ph2dzhdqm8gp1w0azq1h99gjhifw9ncm-darwin-system-22.11pre397212.8c7576622ae+darwin4.0000000.drv' failed to build
```

### Notify maintainers
@allonsy @blaggacao

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"                                                                                                                                                                     1 ↵
these 2 paths will be fetched (0.34 MiB download, 2.06 MiB unpacked):
  /nix/store/w8dik2n00dg8rgqiyjw300xwby01qjzr-bash-interactive-5.1-p16-doc
  /nix/store/z4w3syjmjwzkfvwnq5lgd780ggps2ijg-bash-interactive-5.1-p16-dev
copying path '/nix/store/w8dik2n00dg8rgqiyjw300xwby01qjzr-bash-interactive-5.1-p16-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/z4w3syjmjwzkfvwnq5lgd780ggps2ijg-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.3`
 - channels(retrry): `"darwin, home-manager, nixpkgs"`
 - channels(root): `"darwin, home-manager, nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

--- a/pkgs/tools/misc/git-town/default.nix
+++ b/pkgs/tools/misc/git-town/default.nix
@@ -1,15 +1,24 @@
-{ lib, buildGoModule, fetchFromGitHub, installShellFiles, git, testers, git-town, makeWrapper }:
+{ lib, buildGoModule, fetchFromGitHub, fetchpatch, installShellFiles, git, testers, git-town, makeWrapper }:
 
 buildGoModule rec {
   pname = "git-town";
-  version = "7.7.0";
+  version = "7.8.0";
 
   src = fetchFromGitHub {
     owner = "git-town";
     repo = "git-town";
     rev = "v${version}";
-    sha256 = "sha256-FpBEBx2gb33fGDndvZmvG1A61NoJ4Qy4V3YQSb+Ugsc=";
+    sha256 = "sha256-g9ooMIMN8DN2FcWYkDC1hICCleQYdHf30PYMCit/NMI=";
   };
+
+  patches = [
+    # Fix "go vet" when building using Go 1.18.
+    (fetchpatch {
+      name = "fix-go-vet-in-go-1.18.patch";
+      url = "https://github.com/git-town/git-town/commit/23eb0aca7b28c6a0afc21db553aa0e35d35891aa.patch";
+      sha256 = "sha256-EyfhKVrQxRJNrYqaZI04dJogaXs1J+bbOIu7p8g2Clc=";
+    })
+  ];
 
   vendorSha256 = null;
 

